### PR TITLE
Add BatchAI SDK and tests

### PR DESCRIPTION
**Pull request by llidev**
## Description
Onboarding of .NET SDK for Azure BatchAI Service and related tests. This is being generated by the following spec: https://github.com/Azure/azure-rest-api-specs/tree/current/specification/batchai/resource-manager/Microsoft.BatchAI/2017-09-01-preview. 
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by shahabhijeet** [comment]
@lliimsft please provide a description about the PR.
Describe what changes are you making.
Provide a link to the spec PR

**Comment by lliimsft** [comment]
Hi,

Thanks for the email.
We are Azure Batch AI team. Our PR (Add BatchAI SDK and tests  #3764) includes onboarding of our SDK and related tests. We want to be part of the SDK release that goes out on the 9th.

The link is https://github.com/Azure/azure-sdk-for-net/pull/3764

Can you please help here?

Thanks,
Li

**Comment by lliimsft** [comment]
We have updated the descriptions with info on swagger changes.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.BatchAI</PackageId>
+    <Description>Provides management capabilities for Azure Batch AI.</Description>
+    <VersionPrefix>1.0.0-preview</VersionPrefix>
+    <AssemblyName>Microsoft.Azure.Management.BatchAI</AssemblyName>    
+    <PackageTags>Microsoft Azure BatchAI;BatchAI;REST HTTP client;windowsazureofficial;netcore451511</PackageTags>
```
[the remark this answers is not in the document]
Comment (lliimsft) [reply src/SDKs/BatchAI/Management.BatchAI/Microsoft.Azure.Management.BatchAI.csproj]: we've added note as "This is a public preview release of the Azure BatchAI SDK. Included with this release is support for all cluster, job and file server related APIs."

File: src/SDKs/BatchAI/Management.BatchAI/generate.cmd [hunk 2]
```diff
@@ -0,0 +1,7 @@
+::
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+
+@echo off
+call %~dp0..\..\..\..\tools\generate.cmd batchai/resource-manager %*
```
[the remark this answers is not in the document]
Comment (lliimsft) [reply src/SDKs/BatchAI/Management.BatchAI/generate.cmd]: we've pushed the generated metadata file at \azure-sdk-for-net\src\SDKs\_metadata

File: src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/getmodule.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+var myMoule = require('./singleobject').Hello("dsf");
+
+/* myMoule.setName("lihao__");
```
Comment (shahabhijeet) [inline src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/getmodule.js]: @lliimsft are these files included by mistake?
If not, can you explain what is the purpose of this file for .NET SDK?

File: src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/module.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var name;
```
Comment (shahabhijeet) [inline src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/module.js]: @lliimsft any reasons this file in included in the .NET SDK?

File: src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/singleobject.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+function Hello(){
```
Comment (shahabhijeet) [inline src/SDKs/BatchAI/Management.BatchAI/node_modules/nodejs/1/singleobject.js]: @lliimsft any reasons this file in included in the .NET SDK?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**